### ISSUE TITLE
ANT build fails on JDK 1.8.0+ [styling]

### ISSUE THREAD

**Issue by nickmitchko** [mgarin/weblaf#451]
Multiple ANT builds fail on JDK 1.8.0+ because of a removed sun api.

**Temp Fix:** remove line 345 in [WButtonListener.java](https://github.com/mgarin/weblaf/blob/styling/modules/ui/src/com/alee/laf/button/WButtonListener.java#L345)

```JAVA
@Override
public boolean isEnabled ( final Object sender ){//...}

// After Change =>
public boolean isEnabled ( final Object sender ){//...}
```

**Error:**
```BASH
$ pwd
.../weblaf/build
$ ant build.complete.artifacts
// omitted Log
    [javac] /media/nicholai/AEEE-B5B7/Libraries/Java/WebLAF/1.29-styling/weblaf/modules/ui/src/com/alee/laf/button/WButtonListener.java:345: error: method does not override or implement a method from a supertype
    [javac] 	@Override
    [javac] 	^
// ...
    [javac] 1 error
    [javac] 16 warnings

BUILD FAILED
/media/nicholai/AEEE-B5B7/Libraries/Java/WebLAF/1.29-styling/weblaf/build/build.xml:436: The following error occurred while executing this line:
/media/nicholai/AEEE-B5B7/Libraries/Java/WebLAF/1.29-styling/weblaf/build/build.xml:466: The following error occurred while executing this line:
/media/nicholai/AEEE-B5B7/Libraries/Java/WebLAF/1.29-styling/weblaf/build/build.xml:72: Compile failed; see the compiler error output for details.
```

**Comment by mgarin** [comment]
I tried compiling library with latest JDK8u121 and didn't receive any errors, though there are indeed a few Sun APIs used in the code which might not be present in the future JDK versions, so I will be getting rid of those.

My guess is that you tried compiling sources using Open JDK 8 - in that case some Sun APIs might actually be missing.

**Comment by mgarin** [comment]
I have replaced last few proprietary API usages that were present in the library in [ 912bfd53566895d641109695a0d4bbadb7eb4618 ] commit, you shouldn't get any more related errors.

New artifacts will soon be available in snapshot maven repository:
https://oss.sonatype.org/content/repositories/snapshots/com/weblookandfeel/

**Comment by mgarin** [comment]
Though I'm still not sure why you were getting issue with `@Override` annotation in class extending `UIAction` since `UIAction` has `public boolean isEnabled ( final Object sender )` method on all JDK versions, even Open JDK.

**Comment by nickmitchko** [comment]
It was the issue of openjdk, sorry for not explaining that. Specifically, openjdk is the default Java version on most linux versions. In fact, I could only get the build and library to work with the oracle specific runtime and jdk. The build failed on openjdk versions 7, 8, 9 because of the override.

**Comment by mgarin** [comment]
I see, thanks for the explanation!

The biggest problem there is that `UIAction` class is proprietary API but it is used (indirectly) within `JComponent` in `processKeyBinding ( ... )` method and to implement exactly the same behavior default UIs have I need to use `UIAction` class or fallback to existing action implementations. 

I tried to solve that problem for a few hours yesterday and didn't find any good solution. The one I pushed to git simply uses default `UIAction` implementation and it is bad due to many reasons - unreliable due to proprietary API usage (just less visible, through reflection), inability to change action behavior, overall code mess it creates.

I might actually keep proprietary API usage at this point or, as a different solution, completely rework the way component `ActionMap` is loaded within the components I've fully moved to the brand new UIs.
